documentation/doxygen.py

```diff
@@ -1992,39 +1992,69 @@ def parse_func(state: State, element: ET.Element):
-    # return type, Doxygen swaps the order of those two, causing the constexpr
-    # to be last. See the cpp_function_attributes test for a verification.
-    elif func.type.endswith('constexpr'):
-        func.type = func.type[:-10]
-        func.is_constexpr = True
-    else:
-        func.is_constexpr = False
-    # When 1.8.18 encounters `constexpr static`, it keeps the static there. For
-    # `static constexpr` it doesn't. In both cases the static="yes" is put
-    # there correctly. WHY DOXYGEN, WHY?!
-    if func.type.startswith('static'):
-        func.type = func.type[7:]
+    #
+    # First the prefix keywords - Doxygen has a habit of leaking attributes and
+    # other specifiers into the function's return type, and not necessarily
+    # in any consistent order (including swapping it with the actual type!)
+    func.is_constexpr = False
+    func.is_consteval = False
+    func.is_explicit = False
+    func.is_virtual = False
+    func.is_static = False
+    keywords_behaving_badly = (
+        'constexpr',
+        'consteval',
+        'explicit',
+        'static',
+        'friend',
+        'inline',
+        'virtual'
+    )
+    matched_bad_keyword = True
+    while matched_bad_keyword:
+        matched_bad_keyword = False
+        for kw in keywords_behaving_badly:
+            if func.type == kw: # constructors
+                matched_bad_keyword = True
+                func.type = ''
+            elif func.type.startswith(kw+' '):
+                matched_bad_keyword = True
+                func.type = func.type[len(kw):].strip()
+            elif func.type.endswith(' '+kw):
+                matched_bad_keyword = True
+                func.type = func.type[:len(kw)].strip()
+            else:
+                continue
+            if kw == 'constexpr':
+                func.is_constexpr = True
+            elif kw == 'consteval':
+                func.is_consteval = True
+            elif kw == 'explicit':
+                func.is_explicit = True
+            elif kw == 'virtual':
+                func.is_virtual = True
+            elif kw == 'static':
+                func.is_static = True
+    # Merge any leaked attributes with their corresponding XML attributes to
+    # account for the situation where Doxygen has only half got it right
+    # (which, honestly, is most of the time)
+    func.is_explicit = func.is_explicit or element.attrib['explicit'] == 'yes'
+    func.is_virtual = func.is_virtual or element.attrib['virt'] != 'non-virtual'
+    func.is_static = func.is_static or element.attrib['static'] == 'yes'
+    if 'constexpr' in element.attrib:
+        func.is_constexpr = func.is_constexpr or element.attrib['constexpr'] == 'yes'
+    if 'consteval' in element.attrib:
+        func.is_consteval = func.is_consteval or element.attrib['consteval'] == 'yes'
+    if func.is_consteval:
+        func.is_constexpr = False # consteval trumps constexpr
```

Comment:
Oh and given the theme template updates, you can drop this `if`. (I'm also keeping both `is_override` and `is_final` set, if a function is final.)